lib/plug/csrf_protection.ex

```diff
@@ -106,26 +111,27 @@ defmodule Plug.CSRFProtection do
       raise InvalidCSRFTokenError
     end
 
-    register_before_send(conn, &ensure_same_origin_and_csrf_token(&1, csrf_token))
+    register_before_send(conn, &ensure_same_origin_and_csrf_token!(&1, csrf_token))
   end
 
   ## Verification
 
   defp verified_request?(conn, csrf_token) do
     conn.method in @unprotected_methods
+      || skip_csrf_protection?(conn)
```

Comment:
The previous order was on purpose. We are checking by order that is most likely to match in production. The flag will only match in tests, so we moved to the bottom.